pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -120,6 +124,16 @@ func (p *provider) getConfig(s runtime.RawExtension) (*Config, *providerconfig.C
 		return nil, nil, err
 	}
 
+	c.TemplateNetName, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.TemplateNetName)
```

Comment:
If there is only one network on the templateVM, please default to that in `AddDefaults`